### PR TITLE
[Refactor] Update SystemContext API to support `useContext`

### PR DESCRIPTION
**Pull request by damassi**
Updates `Artsy/SystemContext` to support new hooks context pattern (just needed to export the main context object) and updates most of our functional components.

The new pattern is: 
```js
import { useContext } from 'react'
import { SystemContext } from 'Artsy'

const Foo = props => {
  const contextProps = useContext(SystemContext)
  ... 
}
```

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 7a8832f46e6bfc57c2cf1f5e6bc922aa44be290e

https://deploy-preview-2113--artsy-reaction.netlify.com

**Comment by artsyit** [comment]
:rocket: PR was released in v14.3.0 :rocket:

File: src/Components/v2/Footer.tsx [hunk 1]
```diff
@@ -24,21 +23,16 @@ interface Props {
 }
 
 export const Footer: React.SFC<Props> = props => {
```
[the remark this answers is not in the document]
Comment (damassi) [reply src/Components/v2/Footer.tsx]: Ah, this was an oversight 👍

**Comment by damassi** [comment]
> Of not immediately retrofitting hooks throughout the codebase

I totally agree when it comes to things like class components and state and so on, but with `useContext` it seemed like an easy / safe win :)

**Comment by damassi** [comment]
Welp, not so safe. Looks like we hit an issue with Stitch:

```
Hooks can only be called inside the body of a function component. (https://fb.me/react-invalid-hook-call)
``` 

This could be trivial or not. Going to roll this back while we do more testing.

**Comment by ds300** [comment]
FWIW i think thats the error you get when there's more than one react
instance, and the wrong version of react/react-dom is being used.

**Pull request merged**